cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/wizard/generate/control/GenerateSelectionJob.java

```diff
@@ -84,8 +84,11 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
                     cobigenWrapper.getWorkspaceDependentTemplateDestinationPath(generationReport.getGeneratedFiles());
                 Set<IProject> projects = Sets.newHashSet();
                 for (String filePath : generatedFiles) {
+
                     IProject project =
                         ResourcesPlugin.getWorkspace().getRoot().getProject(PathUtil.getProject(filePath));
+                    project = PathUtil.getRelativeProjectIfNeeded(filePath, project);
```

Comment:
you are overwriting the project variable all the time? Wouldn't that make just sense if the project is null or does not exist?